controllers/cloud.redhat.com/providers/database/singledb.go

```diff
@@ -0,0 +1,261 @@
+		return map[string]string{
+			"hostname": fmt.Sprintf("%v.%v.svc", nn.Name, nn.Namespace),
+			"port":     "5432",
+			"username": utils.RandString(16),
+			"password": utils.RandString(16),
+			"pgPass":   utils.RandString(16),
+			"name":     p.Env.Name,
+		}
+	}
+
+	secMap, err := providers.MakeOrGetSecret(p.Ctx, p.Env, p.Cache, SingleDBSecret, nn, dataInit)
+	if err != nil {
+		return &singleDbProvider{}, errors.Wrap("Couldn't set/get secret", err)
+	}
+
+	dbCfg.Populate(secMap)
+	dbCfg.AdminUsername = "postgres"
+	dbCfg.SslMode = "disable"
+
+	dbp.Config = dbCfg
+
+	var image string
+
+	var dbVersion int32 = 12
+	// if app.Spec.Database.Version != nil {
+	// 	dbVersion = *(app.Spec.Database.Version)
+	// }
+
+	image, ok := imageList[dbVersion]
+
+	if !ok {
+		return &singleDbProvider{}, errors.New(fmt.Sprintf("Requested image version (%v), doesn't exist", dbVersion))
+	}
+
+	imgComponents := strings.Split(image, ":")
+	tag := "cyndi-" + imgComponents[1]
+	image = imgComponents[0] + ":" + tag
+
+	labels := &map[string]string{"sub": "single_db"}
+
+	provutils.MakeLocalDB(dd, nn, p.Env, labels, &dbCfg, image, p.Env.Spec.Providers.Database.PVC, p.Env.Name)
+
+	if err = p.Cache.Update(SingleDBDeployment, dd); err != nil {
+		return &singleDbProvider{}, err
+	}
+
+	s := &core.Service{}
+	if err := p.Cache.Create(SingleDBService, nn, s); err != nil {
+		return &singleDbProvider{}, err
+	}
+
+	provutils.MakeLocalDBService(s, nn, p.Env, labels)
+
+	if err = p.Cache.Update(SingleDBService, s); err != nil {
+		return &singleDbProvider{}, err
+	}
+
+	if p.Env.Spec.Providers.Database.PVC {
+		pvc := &core.PersistentVolumeClaim{}
+		if err := p.Cache.Create(SingleDBPVC, nn, pvc); err != nil {
+			return &singleDbProvider{}, err
+		}
+
+		provutils.MakeLocalDBPVC(pvc, nn, p.Env)
+
+		if err = p.Cache.Update(SingleDBPVC, pvc); err != nil {
+			return &singleDbProvider{}, err
+		}
+	}
+
+	return &singleDbProvider{Provider: *p, Config: dbCfg}, nil
+}
+
+// CreateDatabase ensures a database is created for the given app.  The
+// namespaced name passed in must be the actual name of the db resources
+func (db *singleDbProvider) Provide(app *crd.ClowdApp, c *config.AppConfig) error {
+	if app.Spec.Database.Name == "" && app.Spec.Database.SharedDBAppName == "" {
+		return nil
+	}
+
+	if app.Spec.Database.SharedDBAppName != "" {
+		return db.processSharedDB(app, c)
+	}
+
+	host := db.Config.Hostname
+	port := db.Config.Port
+	user := db.Config.AdminUsername
+	password := db.Config.AdminPassword
+	dbname := app.Spec.Database.Name
+
+	appSqlConnectionString := fmt.Sprintf("host=%s port=%d user=%s password=%s dbname=%s sslmode=disable", host, port, user, password, dbname)
+
+	dbClient, err := sql.Open("postgres", appSqlConnectionString)
+	if err != nil {
+		return err
+	}
+
+	defer dbClient.Close()
+
+	pErr := dbClient.Ping()
+	if pErr != nil {
+		if strings.Contains(pErr.Error(), fmt.Sprintf("database \"%s\" does not exist", app.Spec.Database.Name)) {
```

Comment:
If the database is rebooted - this action will need to happen again right?

Reply:
if the db is rebooted, won't clowder get a reconciliation and can reset the flag?

Reply:
How should it know to reset, I guess we could use some id inside it to say that this is our active db, and so avoid talking to it, unless it's "changed"

Reply:
While it might be nice to check the DB instance, I don't see anything being done with that check besides creating the DB in the instance.

My main concern is halting reconciliations because clowder is hung waiting on the DB connection to return.  Is there a way we can impose a strict timeout on connecting to the DB an executing our SQL?

I think that would be the best solution here because figuring out how to track if the DB got rebooted or not is not trivial.